### ISSUE TITLE
SEP-10 v3.1 Changes (from the SDF)

### ISSUE THREAD

**Issue by JakeUrban** [stellar/go#3334]
### TL;DR

Implement the changes described the [SEP-10 3.1 commit](https://github.com/stellar/stellar-protocol/commit/6c8c9cf6685c85509835188a136ffb8cd6b9c11c).

This likely involves adding an optional argument to the SEP-10 utility functions for adding an additional Manage Data operation containing the domain of the SEP-10 server (`web_auth_domain`).

### Background & Description

Thank you for implementing [SEP-10 v3.0](9d121f9), which reintroduced verification of the home domain of the service requiring SEP-10 authentication, ensuring that signed challenges cannot be used for any other service.

SEP-10 v3.1 adds an optional Manage Data operation that includes the home domain of the service issuing SEP-10 challenges. This allows clients and servers to verify the issuer of the challenge.

**Comment by JakeUrban** [comment]
https://github.com/stellar/stellar-protocol/pull/767

**Comment by leighmcculloch** [comment]
I plan to implement this in next sprint.

**Comment by leighmcculloch** [comment]
This change required for this is ready for review in #3359.
